training/enhanced_dataset.py

```
@@ -0,0 +1,516 @@
+            'mode': self.mode
+        }
+
+        with open(save_path, 'w') as f:
+            json.dump(stats, f, indent=2)
+
+        print(f"Dataset info saved to {save_path}")
+
+
+def create_dataloaders(positive_dir: str,
+                      negative_dir: str,
+                      features_dir: str = None,
```

Ruff (0.13.1)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)